models/topic-model.js

```diff
@@ -21,38 +21,50 @@ const fetchArticleByID = (articleID)=>{
   })
 }
 
-const fetchAllArticles = () =>{
+const fetchAllArticles = () => {
   return db.query(`SELECT 
-  articles.article_id,
-  articles.author,
-  articles.title,
-  topic,
-  articles.created_at,
-  articles.votes,
-  article_img_url,
-  COUNT(comments.article_id) AS comment_count
-FROM
-  articles
-LEFT JOIN
-  comments ON articles.article_id = comments.article_id
-GROUP BY
-  articles.article_id
-ORDER BY
-  created_at DESC;`).then(({rows})=>{
-    return rows
-  })
-}
+      articles.article_id,
+      articles.author,
+      articles.title,
+      topic,
+      articles.created_at,
+      articles.votes,
+      article_img_url,
+      COUNT(comments.article_id) AS comment_count
+    FROM
+      articles
+    LEFT JOIN
+      comments ON articles.article_id = comments.article_id
+    GROUP BY
+      articles.article_id
+    ORDER BY
+      created_at DESC;`)
+    .then(result => {
+      return result.rows;
+    })
+    .catch(err => {
+      next(err);
+    });
+};
 
-const fetchCommentsByArticle = async (article_id) => {
-  const articleExists = await db.query("SELECT COUNT(*) FROM articles WHERE article_id = $1;", [article_id]);
 
-  if (articleExists.rows[0].count === '0') {
-    return Promise.reject({status:404, message: "Article not found"})
-  }
+const fetchCommentsByArticle = (article_id) => {
+  return db.query("SELECT COUNT(*) FROM articles WHERE article_id = $1;", [article_id])
+    .then(articleExistsResult => {
```

Comment:
I would have a look at [these notes](https://notes.northcoders.com/courses/js-back-end/advanced-error-handling) about how to implement this checkExists and make it reusable 👍